src/opnsense/scripts/firmware/bogons.sh

```diff
@@ -73,15 +73,23 @@ LINES_V6=`wc -l ${WORKDIR}/fullbogons-ipv6.txt | awk '{ print $1 }'`
 if [ $BOGONS_V6_TABLE_COUNT -gt 0 ]; then
     ENTRIES_V6=`pfctl -vvsTables | awk '/-\tbogonsv6$/ {getline; print $2}'`
     if [ $ENTRIES_MAX -gt $((2*ENTRIES_TOT-${ENTRIES_V6:-0}+LINES_V6)) ]; then
-        egrep -iv "^fc00::/7" ${WORKDIR}/fullbogons-ipv6.txt > ${DESTDIR}/bogonsv6
+        # private and pseudo-private networks will be excluded
+        # as they are being operated by a separate GUI option
+        cat ${WORKDIR}/fullbogons-ipv6.txt > ${DESTDIR}/bogonsv6
+        echo "!fd00::/8" >> ${DESTDIR}/bogonsv6
+        echo "!fe80::/64" >> ${DESTDIR}/bogonsv6
         RESULT=`/sbin/pfctl -t bogonsv6 -T replace -f ${DESTDIR}/bogonsv6 2>&1`
         echo "$RESULT" | awk '{ print "Bogons V6 file updated: " $0 }' | logger
     else
         echo "Not saving or updating IPv6 bogons (increase table-entries limit)" | logger
     fi
 else
     if [ $ENTRIES_MAX -gt $((2*ENTRIES_TOT+LINES_V6)) ]; then
-        egrep -iv "^fc00::/7" ${WORKDIR}/fullbogons-ipv6.txt > ${DESTDIR}/bogonsv6
+        # private and pseudo-private networks will be excluded
+        # as they are being operated by a separate GUI option
+        cat ${WORKDIR}/fullbogons-ipv6.txt > ${DESTDIR}/bogonsv6
+        echo "!fd00::/8" >> ${DESTDIR}/bogonsv6
+        echo "!fe80::/64" >> ${DESTDIR}/bogonsv6
```

Comment:
```suggestion
        : > ${WORKDIR}/bogonsv6
        echo "!fd00::/8" >> ${WORKDIR}/bogonsv6
        echo "!fe80::/10" >> ${WIRKDIR}/bogonsv6
        cat ${WORKDIR}/fullbogons-ipv6.txt >> ${WORKDIR}/bogonsv6
        mv ${WORKDIR}/bogonsv6 ${DESTDIR}/bogonsv6
```

This then? We can remove a few sins of the past here too by preparing the file and moving it into place after it's ready for an atomic replace.